pkg/aws/metadata/handler_credentials.go

```diff
@@ -56,12 +58,29 @@ func (c *credentialsHandler) Handle(ctx context.Context, w http.ResponseWriter,
 		return http.StatusInternalServerError, fmt.Errorf("error finding pod for ip %s: %s", ip, err.Error())
 	}
 
+	requestedRole := mux.Vars(req)["role"]
+
+	if foundRole == "" && c.defaultRole == requestedRole {
```

Comment:
Is there a reason for this block to interact directly with the metadata API rather than the server? 

If possible I'd rather keep the model simpler and ensure all credential accesses go via the server- so the agent could still determine a default role, but the credentials would still be requested from the server.

Reply:
I added this comment to the start of the block, the code I was referring to is this:

```
resp, err := http.Get(fmt.Sprintf("http://169.254.169.254/latest/meta-data/iam/security-credentials/%s", c.defaultRole))
```